### ISSUE TITLE
Partial Reliability

### ISSUE THREAD

**Issue by aboba** [WICG/web-transport#30]
There are situations intermediate between the "reliable transport" and "no retransmissions" modes specified in the current document.  As an example, a real-time communications application might want to support maxPacketLifetime or maxRetransmissions, so that real-time data would not be delivered after it had become stale.  As an example, the rtx-time parameter in RTX could be emulated over QUIC via maxPacketLifetime applied to a QUIC stream. 

There are ongoing discussions about how to support partial reliability in QUIC. Proposals include:
1. DATAGRAM : https://tools.ietf.org/html/draft-pauly-quic-datagram
2. Partially reliable streams: https://tools.ietf.org/html/draft-lubashev-quic-partial-reliability
3. send message, and then RST_STREAM (if maxPacketLifetime or maxRetransmissions is exceeded): 
https://github.com/quicwg/ops-drafts/issues/15

However, from an API point of view, it doesn't appear difficult to extend createStream() for partial reliability: 

partial dictionary RTCQuicStreamParameters {
    unsigned short maxPacketLifetime ; 
    unsigned short maxRetransmissions; 
};

**Comment by Atrius** [comment]
I've encountered some practical difficulties with the model of abortWriting()/retire() for partial reliability, and I think it's worth considering support for TTL and/or max retransmission count in the API.

I don't believe these feature require any special transport-layer support if they apply to an entire stream:  the implementation simply resets the stream rather than retransmitting anything after the TTL or max retransmission count is exceeded.

I've found it far simpler to implement these kind of features inside a QUIC stream than above it.  The implementation knows when it's about to retransmit data and can evaluate the TTL or max retransmission count at that moment.  The user must keep some mapping of streams to deadlines and schedule callbacks to invoke abortWriting().  And the application needs to know how many retransmissions have happened and when that number changes in order to implement a max retransmission count.

**Comment by aboba** [comment]
@atrius I tend to agree with you. Both maxPacketLifeTime and maxRetransmissions are parameters in the RTCDataChannel API, and neither can be faithfully implemented with application timers.
